plutus-core/plutus-core/src/PlutusCore/Constant/Dynamic/Emit.hs

```diff
@@ -46,3 +46,6 @@ newtype NoEmitterT m a = NoEmitterT
 
 instance Monad m => MonadEmitter (NoEmitterT m) where
     emit _ = pure ()
+
+instance (Monad m, MonadEmitter m) => MonadEmitter (ExceptT e m) where
```

Comment:
Why do we need it now?

Reply:
Extremely bad reasons. `applyTypeSchemed` still wants `MonadError`, and so we effectively run it in `ExceptT CekM` and immediately strip off the `ExceptT`. But then it also wants `MonadEmitter`. We have an instance for `CekM`, and this gives us one for `ExceptT CekM`.

But I couldn't see how to avoid this without making all the CAM use exceptions, because this also goes down into `readKnown` etc.